### PR TITLE
Add advanced filtering panel to history, align appearance of panels

### PR DESCRIPTION
**Pull request by guerler**
Requires #13623. This PR refactors the history header appearance, by permanently displaying the search field (as in the old history and the tool panel). This is achieved mostly through refactoring, if there are conflicts with currently existing additions to the new history panel like e.g. the selections or bulk operation additions, I can help resolve them.

Additionally this PR displays an advanced filtering panel in the history panel. See demo video below.

https://user-images.githubusercontent.com/2105447/160485962-ee27aceb-1b16-45e0-a2b7-479cade0bec3.mov

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: client/src/components/History/CurrentHistory/HistoryFilters.test.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            propsData: {
+                filterText: "",
+                showAdvanced: false,
+            },
+            localVue,
+            stubs: {
+                icon: { template: "<div></div> " },
+            },
+        });
+        expect(wrapper.findAll("[description='advanced filters']").length).toBe(0);
```
Comment (davelopez) [inline client/src/components/History/CurrentHistory/HistoryFilters.test.js]: Nothing too relevant, but consider using *data-test* attributes to provide context to your selectors and isolate them from potential changes.
I don't know if we have a rule for that, probably not, but it's considered best practice to do so.
In this case, the `description` attribute may change and break the test in the future.

File: client/src/components/History/CurrentHistory/HistoryFilters.test.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        await wrapper.setProps({ showAdvanced: true });
+        expect(wrapper.findAll("[description='advanced filters']").length).toBe(1);
```
Comment (davelopez) [inline client/src/components/History/CurrentHistory/HistoryFilters.test.js]: Also, if you are expecting only one element consider using `find` and `exists` for readability.

File: client/src/components/History/CurrentHistory/HistoryFilters.test.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            "[placeholder='created before']": "January 1, 2023",
+        };
+        Object.entries(filterInputs).forEach(([selector, value]) => {
+            const filterInput = wrapper.find(selector);
+            if (filterInput.vm && filterInput.props().type == "text") {
+                filterInput.setValue(value);
+            }
+        });
+        const searchButton = wrapper.findAll("[description='apply filters']");
+        searchButton.trigger("click");
```
Comment (davelopez) [inline client/src/components/History/CurrentHistory/HistoryFilters.test.js]:
```suggestion
        await searchButton.trigger("click");
```
Maybe not critical here, but it's the recommended way of calling trigger in an async context, just for consistency :)

**Review by davelopez — APPROVED** [review]
Awesome! the advanced search panel is a HUGE improvement for discoverability and usability!

I think we can always revisit the rest of the visual design and fine-tune it before the release if folks have other suggestions or ideas.

The failing test under `test_galaxy_packages` also fails locally but it seems totally unrelated.

File: client/src/components/History/CurrentHistory/HistoryFilters.test.js (hunk 2 above)
Comment (guerler) [reply client/src/components/History/CurrentHistory/HistoryFilters.test.js]: Thanks @davelopez. I changed it.

File: client/src/components/History/CurrentHistory/HistoryFilters.test.js (hunk 1 above)
Comment (guerler) [reply client/src/components/History/CurrentHistory/HistoryFilters.test.js]: I like the idea of unifying this, and `data-test` seems to be a good option. Right now, we have too many variations throughout the code base. We should briefly discuss this in the UI meeting and then make a call going forward.

**Comment by davelopez** [comment]
Thank you @guerler, I'm going to try and rebase https://github.com/galaxyproject/galaxy/pull/13555 with these changes

**Pull request merged**